client/json-rpc/src/async_client/tests.rs

```diff
@@ -788,7 +788,9 @@ async fn test_batch_send_requests_and_response_id_type_not_matched_error() {
             .batch_send(vec![Request::get_metadata(), Request::get_currencies()])
             .await;
 
-        assert_eq!("Err(UnexpectedError(InvalidResponseIdType(JsonRpcResponse { diem_chain_id: 4, diem_ledger_version: 1, diem_ledger_timestampusec: 1602888396000000, jsonrpc: \"2.0\", id: Some(String(\"1\")), result: Some(Object({\"chain_id\": Number(4), \"timestamp\": Number(234234), \"version\": Number(1)})), error: None })))", format!("{:?}", &err));
+        assert!(
+            format!("{:?}", &err).contains("UnexpectedError(InvalidResponseIdType(JsonRpcResponse")
+        );
         assert_err!(err, Error::UnexpectedError { .. }, true);
```

Comment:
would recommend you merge these lines instead of doing string comparison

something like
```
assert_err!(err, Error::UnexpectedError(UnexpectedError::InvalidResponseIdType { .. })), true);
```

Reply:
tried, won't work. I could not figure out a way to match underlining type, not sure what I did wrong.